src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+    this.class = 'herb';
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.class = 'carn';
+  }
+
+  bite(herb) {
+    if (!herb.hidden && herb.class !== 'carn') {
```

Comment:
The condition should also check if 'herb' is an instance of Herbivore to ensure type safety. Currently, any object with a 'hidden' property and a 'class' property not equal to 'carn' could be passed to the bite method.